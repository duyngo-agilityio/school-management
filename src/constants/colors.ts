const bluePallet = {
  zodiac: '#152259',
  koamaru: '#13296A',
  havelock: '#1C5A8C',
  cornflower: '#3182ce',
  solitude: '#EBF6FF',
  catSkill: '#E2E8F0',
  linkWater: '#EFF3FA',
};

const grayPallet = {
  cod: '#1A1A1A',
  tundora: '#424242',
  emperor: '#4F4F4F',
  ash: '#8A8A8A',
  slate: '#595959',
  aluminum: '#BDBDBD',
};

const whitePallet = {
  seashell: '#F1F1F1',
  snow: '#FCFAFA',
  pure: '#FFFFFF',
};

const redPallet = {
  roman: '#DB5050',
};

const yellowPallet = {
  anzac: '#DBBD50',
};

export const customColors = {
  ...bluePallet,
  ...whitePallet,
  ...grayPallet,
  ...redPallet,
  ...yellowPallet,
} as const;
