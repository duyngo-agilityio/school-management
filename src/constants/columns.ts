const COLUMNS = {
  FIELDS: {
    NAME: 'fullName',
    ID: 'id',
    EMAIL: 'email',
    CLASS: 'class',
    GENDER: 'gender',
    SUBJECT: 'subject',
  },
  HEADER_NAME: {
    NAME: 'Name',
    SUBJECT: 'Subject',
    CLASS: 'Class',
    EMAIL: 'Email Address',
    GENDER: 'Gender',
    STUDENT_ID: 'Student ID',
  },
};

export const COLUMNS_STUDENT = [
  {
    field: COLUMNS.FIELDS.NAME,
    headerName: COLUMNS.HEADER_NAME.NAME,
  },
  {
    field: COLUMNS.FIELDS.ID,
    headerName: COLUMNS.HEADER_NAME.STUDENT_ID,
  },
  {
    field: COLUMNS.FIELDS.EMAIL,
    headerName: COLUMNS.HEADER_NAME.EMAIL,
  },
  {
    field: COLUMNS.FIELDS.CLASS,
    headerName: COLUMNS.HEADER_NAME.CLASS,
  },
  {
    field: COLUMNS.FIELDS.GENDER,
    headerName: COLUMNS.HEADER_NAME.GENDER,
  },
];

export const COLUMNS_TEACHER = [
  {
    field: COLUMNS.FIELDS.NAME,
    headerName: COLUMNS.HEADER_NAME.NAME,
  },
  {
    field: COLUMNS.FIELDS.SUBJECT,
    headerName: COLUMNS.HEADER_NAME.SUBJECT,
  },
  {
    field: COLUMNS.FIELDS.CLASS,
    headerName: COLUMNS.HEADER_NAME.CLASS,
  },
  {
    field: COLUMNS.FIELDS.EMAIL,
    headerName: COLUMNS.HEADER_NAME.EMAIL,
  },
  {
    field: COLUMNS.FIELDS.GENDER,
    headerName: COLUMNS.HEADER_NAME.GENDER,
  },
];
