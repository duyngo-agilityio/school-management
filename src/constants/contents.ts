export const NOTFOUND = {
  STUDENT: {
    TITLE: 'No Students at this time',
    DESCRIPTION: 'Students will appear here after they enroll in your school.',
    BACK_BUTTON: 'Back to Student List'
  },
  TEACHER: {
    TITLE: 'No Teachers at this time',
    DESCRIPTION: 'Teachers will appear here after they enroll in your school.',
    BACK_BUTTON: 'Back to Teacher List'
  },
};
