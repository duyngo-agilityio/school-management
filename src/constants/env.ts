export const BASE_URL = process.env.NEXT_PUBLIC_BASE_URL!;
export const IMG_KEY = process.env.NEXT_PUBLIC_IMG_KEY;
