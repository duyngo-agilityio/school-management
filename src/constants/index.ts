export * from './path';
export * from './placeholder';
export * from './colors';
export * from './columns';
export * from './messages';
export * from './options';
export * from './regex';
export * from './contents';
export * from './routes';
