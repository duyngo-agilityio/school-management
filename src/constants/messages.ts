export const VALIDATE_MESSAGE = {
  EMPTY: 'The input is required',
  EMAIL: 'Invalid email',
  PASSWORD:
    'Password must have at least 8 characters, 1 special character and 1 number',
  REPEAT_PASSWORD: 'Password not matching',
  PHONE_NUMBER: 'Invalid phone number format',
  USERNAME_PASSWORD: 'Email or password invalid. Please help to try again!',
};

export const SUCCESS_MESSAGES = {
  ADD_TEACHER: 'Add Teacher success!',
  ADD_STUDENT: 'Add Student success!',
  DELETE_TEACHER: 'Delete Teacher success!',
  DELETE_STUDENT: 'Delete Student success!',
  EDIT_TEACHER: 'Edit Teacher success!',
  EDIT_STUDENT: 'Edit Student success!',
  SIGNIN_SUCCESS: 'Signed In successfully!',
  SIGNUP_SUCCESS: 'Signed Up successfully!',
};

export const ERROR_MESSAGES = {
  ADD_TEACHER: 'Add Teacher failed!',
  ADD_STUDENT: 'Add Student failed!',
  DELETE_TEACHER: 'Delete Teacher failed!',
  DELETE_STUDENT: 'Delete Student failed!',
  EDIT_TEACHER: 'Edit Teacher failed!',
  EDIT_STUDENT: 'Edit Student failed!',
  PAGE_NOT_FOUND: 'Page not found!',
};

export const ERROR_API = {
  END_POINT: 'Something went wrong!!!',
  NOT_FOUND: 'Not found',
};
