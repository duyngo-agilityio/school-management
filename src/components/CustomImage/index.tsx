'use client';

import Image, { ImageProps } from 'next/image';
import { memo, useState } from 'react';

// Constants
import { NO_IMAGE, NOT_FOUND_IMAGE } from '@/constants';

const CustomImage = ({ src, alt, blurDataURL, ...rest }: ImageProps) => {
  const [fallbackSrc, setFallbackSrc] = useState(false);

  const handleOnError = () => setFallbackSrc(true);
  const altImage = src !== NOT_FOUND_IMAGE && alt ? alt : NO_IMAGE;

  return (
    <Image
      src={fallbackSrc ? NOT_FOUND_IMAGE : src}
      alt={altImage}
      blurDataURL={blurDataURL}
      placeholder="blur"
      onError={handleOnError}
      {...rest}
    />
  );
};

export default memo(CustomImage);
