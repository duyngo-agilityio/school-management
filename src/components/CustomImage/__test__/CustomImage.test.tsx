import { render } from '@testing-library/react';

// Components
import CustomImage from '..';

// Constants
import { BLUR_DATA_URL, NO_IMAGE, NOT_FOUND_IMAGE } from '@/constants';

describe('CustomImage Component', () => {
  it('should match snapshot', () => {
    const { container } = render(
      <CustomImage
        width={50}
        height={50}
        src="/logo.png"
        alt="Test Image"
        blurDataURL={BLUR_DATA_URL}
        placeholder="blur"
      />,
    );

    expect(container).toMatchSnapshot();
  });

  it('should display NO_IMAGE as alt text when src is NOT_FOUND_IMAGE', () => {
    const { getByAltText } = render(
      <CustomImage
        width={50}
        height={50}
        src={NOT_FOUND_IMAGE}
        alt="Test Image"
        blurDataURL={BLUR_DATA_URL}
        placeholder="blur"
      />,
    );

    const image = getByAltText('No Image') as HTMLImageElement;
    expect(image.alt).toBe(NO_IMAGE);
  });
});
