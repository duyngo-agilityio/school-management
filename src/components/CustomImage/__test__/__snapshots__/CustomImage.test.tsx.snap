// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomImage Component should match snapshot 1`] = `
<div>
  <img
    alt="Test Image"
    data-nimg="1"
    decoding="async"
    height="50"
    loading="lazy"
    src="http://localhost/_next/image?url=%2Flogo.png&w=128&q=75"
    srcset="/_next/image?url=%2Flogo.png&w=64&q=75 1x, /_next/image?url=%2Flogo.png&w=128&q=75 2x"
    style="color: transparent; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
    width="50"
  />
</div>
`;
